### ISSUE TITLE
Can't run tutorial under Debian 10 in WSL2

### ISSUE THREAD

**Issue by tig** [datapane/datapane#43]
**Describe the bug**
New user; just signed up. Logged into **https://datapane.com/home/** and followed instructions.

**To Reproduce**

1. Start Debian 10 under WSL2 on Windows 10
2. goto https://datapane.com/home/
3. Copy script to `rept.py`
4. Do this:

```
sudo apt update
sudo apt install -y python3-pip
sudo apt install build-essential libssl-dev libffi-dev python3-dev
pip3 install datapane
python3 ./rept.py
```

4. Get error

**Expected behavior**
Per the instructions, I should be able to refresh that page and see results. 

**Screenshots**
Error from running `rept.py`

```python
Connected successfully to https://datapane.com as cek
Publishing report and associated data - please wait..
Traceback (most recent call last):
  File "./rept.py", line 26, in <module>
    r.publish(name=f'COVID-19 in {" ".join(locations.iso_code)}', open=True, tweet=True, description=f'COVID-19 in {", ".join(locations.location)}')
  File "/home/tig/.local/lib/python3.7/site-packages/datapane/client/api/report.py", line 384, in publish
    res = Resource(self.endpoint).post_files(dict(attachments=attachments), document=report_str, **kwargs)
  File "/home/tig/.local/lib/python3.7/site-packages/datapane/client/api/common.py", line 204, in post_files    e = MultipartEncoder(fields=fields)
  File "/home/tig/.local/lib/python3.7/site-packages/requests_toolbelt/multipart/encoder.py", line 125, in __init__
    self._prepare_parts()
  File "/home/tig/.local/lib/python3.7/site-packages/requests_toolbelt/multipart/encoder.py", line 246, in _prepare_parts
    self.parts = [Part.from_field(f, enc) for f in self._iter_fields()]
  File "/home/tig/.local/lib/python3.7/site-packages/requests_toolbelt/multipart/encoder.py", line 246, in <listcomp>
    self.parts = [Part.from_field(f, enc) for f in self._iter_fields()]
  File "/home/tig/.local/lib/python3.7/site-packages/requests_toolbelt/multipart/encoder.py", line 495, in from_field
    return cls(headers, body)
  File "/home/tig/.local/lib/python3.7/site-packages/requests_toolbelt/multipart/encoder.py", line 488, in __init__
    self.len = len(self.headers) + total_len(self.body)
  File "/home/tig/.local/lib/python3.7/site-packages/requests_toolbelt/multipart/encoder.py", line 432, in total_len
    if hasattr(o, 'len'):
  File "/home/tig/.local/lib/python3.7/site-packages/requests_toolbelt/multipart/encoder.py", line 573, in len
    return total_len(self.fd) - self.fd.tell()
  File "/home/tig/.local/lib/python3.7/site-packages/requests_toolbelt/multipart/encoder.py", line 441, in total_len
    return os.fstat(fileno).st_size
FileNotFoundError: [Errno 2] No such file or directory
```

**Desktop (please complete the following information):**

- OS: Windows 10
- Browser: Edge
- Python: 3.7.3

**Comment by mands** [comment]
Hi @tig  - thanks for raising. 

That's an odd issue - unsure what's causing that but will try in a VM and see if can repro. 

I wonder if there may be something with WSL2 file handling - is the home drive perhaps shared between Windows and WSL?

**Comment by mands** [comment]
Unfortunately I'm unable to repro this on my Linux install (Fedora 33) - couple of questions,
- is the drive shared between Windows and WSL?
- have you tried any other distros in WSL?

**Comment by votti** [comment]
Hi,

I have experienced the same error with Ubuntu 20.04.1 LTS in WSL2 when running the example on Jupyter in a folder located in the mounted Windows file system.

Runnig it from Jupyter in the Ubuntu partition fixes the issue for me.

**Comment by mands** [comment]
Ok - thanks for raising - will look into reproducing and resolving

**Comment by shahinrostami** [comment]
Hi @tig and @votti! Are you still encountering this issue on the latest version of Datapane?

**Comment by shahinrostami** [comment]
Let's close this for now! We can re-open it if the problem emerges again.

**Comment by gabiLembaTherapeutics** [comment]
This happens to me as well when using a VS Code devcontainer. Report upload works fine when running the app outside of the devcontainer.

``` FileNotFoundError: [Errno 2] No such file or directory ```

Exporting the report in HTML format still works fine.

Stack trace:
``` 
    report.upload(name=run_id)
 File "/home/vscode/.local/lib/python3.7/site-packages/datapane/client/api/report/core.py", line 391, in upload
    **kwargs,
File "/home/vscode/.local/lib/python3.7/site-packages/datapane/client/api/report/core.py", line 343, in _upload_report
    res = Resource(self.endpoint).post_files(files, overwrite=overwrite, document=report_str, **kwargs)
File "/home/vscode/.local/lib/python3.7/site-packages/datapane/client/api/common.py", line 202, in post_files
    e = MultipartEncoder(fields=fields)
File "/home/vscode/.local/lib/python3.7/site-packages/requests_toolbelt/multipart/encoder.py", line 125, in __init__
    self._prepare_parts()
File "/home/vscode/.local/lib/python3.7/site-packages/requests_toolbelt/multipart/encoder.py", line 246, in _prepare_parts
    self.parts = [Part.from_field(f, enc) for f in self._iter_fields()]
File "/home/vscode/.local/lib/python3.7/site-packages/requests_toolbelt/multipart/encoder.py", line 246, in <listcomp>
    self.parts = [Part.from_field(f, enc) for f in self._iter_fields()]
File "/home/vscode/.local/lib/python3.7/site-packages/requests_toolbelt/multipart/encoder.py", line 495, in from_field
    return cls(headers, body)
File "/home/vscode/.local/lib/python3.7/site-packages/requests_toolbelt/multipart/encoder.py", line 488, in __init__
    self.len = len(self.headers) + total_len(self.body)
File "/home/vscode/.local/lib/python3.7/site-packages/requests_toolbelt/multipart/encoder.py", line 432, in total_len
    if hasattr(o, 'len'):
File "/home/vscode/.local/lib/python3.7/site-packages/requests_toolbelt/multipart/encoder.py", line 573, in len
    return total_len(self.fd) - self.fd.tell()
File "/home/vscode/.local/lib/python3.7/site-packages/requests_toolbelt/multipart/encoder.py", line 441, in total_len
    return os.fstat(fileno).st_size
```

**Comment by mands** [comment]
This should be fixed in the 0.16.x releases which changed how we handle file processing.
